main.go

```diff
@@ -0,0 +1,147 @@
+package main
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"net/url"
+	"os"
+	"os/signal"
+	"syscall"
+
+	humanize "github.com/dustin/go-humanize"
+	flag "github.com/jessevdk/go-flags"
+
+	"github.com/ishanjain28/pluto/pluto"
+)
+
+var Version string
+var Build string
+
+var options struct {
+	Verbose      bool     `long:"verbose" description:"Enable Verbose Mode"`
+	Connections  uint     `short:"n" long:"connections" default:"1" description:"Number of concurrent connections"`
+	Name         string   `long:"name" description:"Path or Name of save file"`
+	LoadFromFile string   `short:"f" long:"load-from-file" description:"Load URLs from a file"`
+	Headers      []string `short:"H" long:"Headers" description:"Headers to send with each request. Useful if a server requires some information in headers"`
+	Version      bool     `short:"v" long:"version" description:"Print Pluto Version and exit"`
+}
+
+func main() {
+
+	sig := make(chan os.Signal, 1)
+	signal.Notify(sig, os.Interrupt, syscall.SIGINT, syscall.SIGTERM)
+	ctx, cancel := context.WithCancel(context.Background())
+
+	go func() {
+		<-sig
+		fmt.Printf("Interrupt Detected, Shutting Down.")
+		cancel()
+		os.Exit(1)
+	}()
+
+	args, err := flag.ParseArgs(&options, os.Args)
+	if err != nil {
+		fmt.Printf("%s", err.Error())
+		return
+	}
+
+	if options.Version {
+		fmt.Println("Pluto - A Fast Multipart File Downloader")
+		fmt.Printf("Version: %s\n", Version)
+		fmt.Printf("Build: %s\n", Build)
+		return
+	}
+
+	args = args[1:]
+
+	urls := []string{}
+
+	if options.LoadFromFile != "" {
+		f, err := os.OpenFile(options.LoadFromFile, os.O_RDONLY, 0x444)
+		if err != nil {
+			log.Fatalf("error in opening file %s: %v\n", options.LoadFromFile, err)
+		}
+		defer f.Close()
+		reader := bufio.NewReader(f)
+
+		for {
+			str, err := reader.ReadString('\n')
+			if err != nil {
+				if err == io.EOF {
+					break
+				}
+				log.Fatalf("error in reading file: %v\n", err)
+			}
+			u := str[:len(str)-1]
+			if u != "" {
+				urls = append(urls, u)
+			}
+		}
+
+		fmt.Printf("Queued %d urls\n", len(urls))
+	} else {
+		for _, v := range args {
+			if v != "" && v != "\n" {
+				urls = append(urls, v)
+			}
+		}
+
+	}
+	if len(urls) == 0 {
+		log.Fatalf("nothing to do. Please pass some url to fetch")
+	}
+
+	if options.Connections == 0 {
```

Comment:
Using a default value of 16/24/32 instead of printing an error when user doesn't gives a value for number of connections would be better idea

Reply:
I added the default in the options struct:
+	Connections  uint     `short:"n" long:"connections" default:"1" description:"Number of concurrent connections"`

this check is trying to catch an user that passes a 0 connections argument. I set the default to 1, we can change it to the value you think is more appropiate

Reply:
Okay. I hadn't thought of that.

Reply:
Okay. I hadn't thought of that.